### PR TITLE
New constructors for Node

### PR DESCRIPTION
**Pull request by staleyLANL**
Added Node tests for the new constructors.
Renamed some files and directories, to match the class names involved.
Updated relevant cmake-related file to reflect the name changes.
Added or improved some comments. Made others consistent between similar files.
Tweaked some diagnostic-message verbiage.
Minor changes in some variable names.
Improved and shortened some Node add() material; and a bit of new capabilities.
Augmented Node add() test appropriately.

### CODE REVIEW

File: src/GNDStk/Node/src/call.hpp [hunk 1]
```diff
@@ -12,9 +12,9 @@
 // Those encode (via their type) whether we're accessing metadata or children.
 
 // Note: for variant-based meta_t and child_t keywords, you'll still need to
-// call meta<RESULT>() or child<RESULT> directly. Implementing the variant
-// versions here would bulk up this file, and you'd need to write the operator
-// call in functional form, e.g. mynode.operator()<RESULT>(...), to use it.
+// call meta<TYPE>() or child<TYPE> directly. Implementing the variant versions
```
Comment (jlconlin) [inline src/GNDStk/Node/src/call.hpp:15]: I'm not sure `TYPE` is that much better than `RESULT` here. It's not a reason to reject this Pull Request, but I'm wondering why the change.

Comment (staleyLANL) [reply src/GNDStk/Node/src/call.hpp:15]: I originally used RESULT to mean, "result of the query." (As in, the type of the result, given the type system I've put together for this.)

But the system works not just for queries [or existing GNDS hierarachies], but for building brand-new hierarachies. That is to say, you can both read and write.

Given that it works both ways, I decided that the general and agnostic term TYPE was better than the term RESULT.

File: src/GNDStk/Node/test/ctor.test.cpp [hunk 2]
```diff
@@ -3,19 +3,211 @@
...
+      Node<> n(plain::child::reactionSuite);
+      CHECK(n.name == "reactionSuite");
+      CHECK(n.metadata.size() == 0);
+      CHECK(n.children.size() == 0);
+   }
+
+   // ------------------------
+   // child_t<void>, Node
+   // ------------------------
+   {
```
[the remark this answers is not in the document]
Comment (staleyLANL) [reply src/GNDStk/Node/test/ctor.test.cpp:135]: I could probably make global changes, across various test files, to conform more to this style, if this style is considered to be better. How about I make a note of this, but make these changes as a separate PR specifically for this. That way, we can get this PR out-the-door.

File: src/GNDStk/Node/test/ctor.test.cpp [hunk 3]
```diff
@@ -3,19 +3,211 @@
...
+
+   // ------------------------
    // default
-   Node<> n1;
-   CHECK(n1.empty());
+   // ------------------------
+   {
+      Node<> n;
+      CHECK(n.empty());
+   }
```
[the remark this answers is not in the document]
Comment (staleyLANL) [reply src/GNDStk/Node/test/ctor.test.cpp:79]: Same reply as before... I'll make a note of this, but how about doing restyled tests as a separate PR.

File: src/GNDStk/node2type/src/detail.hpp [hunk 4]
```diff
@@ -8,27 +8,25 @@ namespace detail {
...
+   // Our goal here is to extract the "..." - the metadatum's value string.
    for (auto &c : node.children)
       if (c->name == "pcdata")
          for (auto &m : c->metadata)
             if (m.first == "text")
                return m.second;
 
    log::error(
-      "Unable to find pcdata+text entry in the Node,\n"
+      "Internal error in detail::get_pcdata_string().\n"
```
[the remark this answers is not in the document]
Comment (staleyLANL) [reply src/GNDStk/node2type/src/detail.hpp:23]: This is another thing I'm thinking would be best as another PR. At some point I'd like to revisit every diagnostic message, and give them good and consistent verbiage and such across-the-board. This could include trying to make the main messages be one-liners, with additional remarks given as informational statements.

**Comment by staleyLANL** [comment]
As I've seen before, github says I'm "not authorized to merge this pull request." Perhaps we should get me such authorization, so that I can merge something after changes have been approved by one or more reviewers. For now, feel free to merge this @jlconlin.
